src/hmm/hmm-topology.cc

```diff
@@ -177,11 +199,13 @@ void HmmTopology::Write(std::ostream &os, bool binary) const {
   } else {
     WriteIntegerVector(os, binary, phones_);
     WriteIntegerVector(os, binary, phone2idx_);
+    if (!is_hmm) WriteBasicType(os, binary, static_cast<int32>(-1));
```

Comment:
comment that this -1 is a signal that the object has the new, extended format with SelfLoopPdfClass.